packages/plugin-node-modules/sources/NodeModulesLinker.ts

```diff
@@ -582,7 +582,9 @@ async function persistNodeModules(preinstallState: NodeModulesLocatorMap | null,
   }
 
   const progress = Report.progressViaCounter(addList.length);
-  report.reportProgress(progress);
+  if (addList.length === 0)
+    // Edge case - zero packages to install, - do not report progress
+    report.reportProgress(progress);
```

Comment:
Shouldn't that rather be `addList.length !== 0`? In any case, I feel like it might be better if we can make `StreamReport` recognize this - maybe it's just a matter of checking whether the generator is done within `reportProgress`.

Reply:
@arcanis On a closer look the `StreamReport` and `progressViaCounter` do everything right. The bug is that I forgot to tick on workspaces (which are not being really added). Pushed the fix